### ISSUE TITLE
Question about "Convert a training-time RepViT into the inference-time structure" with replace_batchnorm.

### ISSUE THREAD

**Issue by Topdu** [THU-MIG/RepViT#39]
@jameslahm Hi,

Thanks for your great work !!
The model outputs obtained using replace_batchnorm are inconsistent, which is puzzling, as to how to get the correct results:
```python
    model = create_model(model_name, num_classes=1000)
    y = model(inputs)
    print(y)
    utils.replace_batchnorm(model)
    y = model(inputs)
    print(y)
```
result:
tensor([[ 0.0716, -0.1728, -0.6425,  ..., -0.1067, -1.0730, -0.2081],
        [-0.0716,  0.1728,  0.6425,  ...,  0.1067,  1.0730,  0.2081]])
tensor([[-0.0643, -0.4197,  0.1459,  ..., -0.0541, -0.1685,  0.1784],
        [-0.0643, -0.4197,  0.1459,  ..., -0.0541, -0.1685,  0.1784]])

**Comment by jameslahm** [comment]
Do the modules in the model have a correct implementation of the `fuse` function? Please note that the BatchNormalization in the module without `fuse` function will be simply replaced with `nn.Identity`.

**Comment by Topdu** [comment]
Thank you for the reply, the code used here is cloned from your repo without any changes.

**Comment by Topdu** [comment]
I just modified speed_gpu.py as follows:
```python
from argparse import ArgumentParser

parser = ArgumentParser()

parser.add_argument('--model', default='repvit_m0_9', type=str)
parser.add_argument('--resolution', default=224, type=int)
parser.add_argument('--batch-size', default=2, type=int)

if __name__ == "__main__":
    args = parser.parse_args()
    model_name = args.model
    batch_size = args.batch_size
    resolution = args.resolution
    torch.cuda.empty_cache()
    inputs = torch.randn(batch_size, 3, resolution,
                            resolution)
    model = create_model(model_name, num_classes=1000)
    y = model(inputs)
    print(y)
    utils.replace_batchnorm(model)
    y = model(inputs)
    print(y)
    # model.to(device)
    # model.eval()
    # throughput(model_name, model, device, batch_size, resolution=resolution)
```

**Comment by Topdu** [comment]
It's working! Thanks!

**Comment by dlml** [comment]
When using "repalce_batchnorm", the result of converting to **onnx** format is inconsistent @Topdu @jameslahm

**Comment by seermer** [comment]
I also found some inconsistency after `replace_batchnorm`

```
x = torch.randn(4, 3, 224, 224, requires_grad=False).cuda()
model = timm.create_model("repvit_m2_3.dist_450e_in1k",
                          pretrained=True, num_classes=0).cuda()
model.eval()
output = model(x)

replace_batchnorm(model)
model.eval()
output_fused = model(x)

print(output)
print(output_fused)
```

gives

> tensor([[ 0.7834,  0.1387, -1.7876,  ...,  0.3591,  2.0152, -2.6656],
>         [ 0.8065,  0.1328, -1.8703,  ...,  0.3418,  2.0391, -2.6092],
>         [ 0.8388,  0.1825, -1.8846,  ...,  0.3555,  2.0984, -2.6080],
>         [ 0.8539,  0.1291, -1.8841,  ...,  0.3301,  2.0132, -2.5867]],
>        device='cuda:0')
> tensor([[ 0.7828,  0.1394, -1.7882,  ...,  0.3586,  2.0155, -2.6663],
>         [ 0.8061,  0.1330, -1.8712,  ...,  0.3412,  2.0409, -2.6105],
>         [ 0.8384,  0.1824, -1.8850,  ...,  0.3548,  2.0999, -2.6089],
>         [ 0.8536,  0.1293, -1.8849,  ...,  0.3293,  2.0152, -2.5878]],
>        device='cuda:0')

the difference is in about 1e-3, which is higher than usual accepted margin of error (1e-5 to 1e-7)
